File: testdata/parity/synthetic/go/widget/widget_test.go
package widget

import "testing"

func TestAdd(t *testing.T) {
	if Add(1, 2) != 3 {
		t.Fatal("1+2 != 3")
	}
}

func TestValidate(t *testing.T) {
	o := OrderRequest{ID: "1"}
	if err := o.Validate(); err != nil {
		t.Fatal(err)
	}
}

func TestProcessAll(t *testing.T) {
	ProcessAll() // just exercises the cross-file caller
}

File: testdata/parity/synthetic/typescript/src/widget.test.ts
import { add, Widget, processAll } from "./widget";

export function testAdd(): void {
  if (add(1, 2) !== 3) throw new Error("fail");
}

export function testValidate(): void {
  const w = new Widget("x", 1);
  if (!w.validate()) throw new Error("fail");
}

export function testProcessAll(): void {
  processAll();
}
